streams/src/main/java/org/apache/kafka/streams/state/internals/StoreChangeLogger.java

```diff
@@ -109,7 +109,7 @@ public void logChange(ValueGetter<K, V> getter) {
             }
             for (K k : this.dirty) {
                 V v = getter.get(k);
-                collector.send(new ProducerRecord<>(this.topic, this.partition, k, v), keySerializer, valueSerializer);
+                collector.send(new ProducerRecord<>(this.topic, this.partition, context.timestamp(), k, v), keySerializer, valueSerializer);
```

Comment:
Hi @guozhangwang perhaps we can just set `log.segment.ms` to `Long.MAX_VALUE` and then rolling will be based on `log.segment.bytes` only?